src/libraries/System.Net.Connections/src/System/Net/Connections/ConnectionExtensions.cs

```diff
@@ -0,0 +1,265 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Connections
+{
+    /// <summary>
+    /// Extension methods for working with the System.Net.Connections types.
+    /// </summary>
+    public static class ConnectionExtensions
+    {
+        /// <summary>
+        /// Retrieves a Type-based property from an <see cref="IConnectionProperties"/>, if it exists.
+        /// </summary>
+        /// <typeparam name="T">The type of the property to retrieve.</typeparam>
+        /// <param name="properties">The connection properties to retrieve a property from.</param>
+        /// <param name="property">If <paramref name="properties"/> contains a property of type <typeparamref name="T"/>, receives the property. Otherwise, default.</param>
+        /// <returns>If <paramref name="properties"/> contains a property of type <typeparamref name="T"/>, true. Otherwise, false.</returns>
+        public static bool TryGet<T>(this IConnectionProperties properties, [MaybeNullWhen(false)] out T property)
```

Comment:
I notice some TryGet variations use "MaybeNullWhen(false)" and some use "NotNullWhen(true)" -- do these mean different things? Why are some one and some the other?

Reply:
`NotNullWhen(true)` does not work when T is non-nullable. @stephentoub recommended `MaybeNullWhen(false)` as a workaround that works with non-nullable types.

I don't 100% understand the intricacies of this feature yet myself. I can't really tell you what the difference between the two is, other than one works here and the other doesn't. @buyaa-n can you help?

Reply:
See the second bullet under https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/api-guidelines/nullability.md#nullable-attributes

Reply:
in short `NotNullWhen(bool)` used for reference types, MaybeNullWhen(bool) used for generic types where it can be a value type where it would never be null